vizro-core/tests/unit/vizro/models/test_base.py

```diff
@@ -151,6 +156,27 @@ def test_add_type_dict_instantiation(self, ParentWithForwardRef, mocker):
         assert isinstance(parent.child, ChildZ)
 
 
+class TestChildForwardRefDiscriminatedUnion:
+    def test_no_type_match(self, Parent):
+        child = ChildWithForwardRef()
+        with pytest.raises(ValidationError, match="No match for discriminator 'type' and value 'child_forward_ref'"):
+            Parent(child=child)
+
+    def test_add_type_model_instantiation(self, Parent, mocker):
+        # Make it as if these are in vizro.models so that update_forward_refs call in add_type works on them.
+        mocker.patch.dict(vm.__dict__, {"ChildXForwardRef": ChildX})
+        Parent.add_type("child", ChildWithForwardRef)
+        parent = Parent(child=ChildWithForwardRef(grandchild=ChildX()))
+        assert isinstance(parent.child, ChildWithForwardRef)
+
+    def test_add_type_dict_instantiation(self, Parent, mocker):
+        # Make it as if these are in vizro.models so that update_forward_refs call in add_type works on them.
+        mocker.patch.dict(vm.__dict__, {"ChildXForwardRef": ChildX})
+        Parent.add_type("child", ChildWithForwardRef)
+        parent = Parent(child={"type": "child_forward_ref", "grandchild": {"type": "child_x"}})
```

Comment:
```suggestion
        parent = Parent(child={"type": "child_forward_ref", "grandchild": {}})
```
We shouldn't need to specify `type` here. If this change doesn't work then please tell me though because it means my understanding of something is wrong...

Reply:
Ah yes, that works!